lib/sidebar_builder.rb

```diff
@@ -0,0 +1,127 @@
+      item_class = item_class(path)
+      child_markup = generate_child_markup(path, page, child_pages, 2)
+      grandchildren_markup = generate_grandchildren_markup(child_pages)
+
+      "<li class=\"#{item_class}\">
+        <div class=\"nav-group-heading\">
+          <i class=\"material-icons\">arrow_right</i>
+          <span>#{title}</span>
+        </div>
+        <ul class=\"nav-ul\">
+          #{child_markup}
+          #{grandchildren_markup}
+        </ul>
+      </li>"
+    end
+
+    def item_class(path)
+      active = @filename.active?(path)
+      return 'nav-group active' if active
+
+      'nav-group'
+    end
+
+    def generate_child_markup(path, page, all_child_pages, level)
+      title = page[:title].split('–').last.strip
+
+      url = page[:url]
+      headers = page[:headers]
+
+      grandchildren = all_child_pages.select do |grandchild_path, _|
+        grandchild_path.include?(path) \
+        && grandchild_path != path \
+        && path.split('/').length > level
+      end
+
+      sub_group_markup = ''
+      has_child_pages = !all_child_pages.empty?
+
+      if (!headers.nil? && headers.any?) || !grandchildren.empty?
+        if has_child_pages
+          active = @filename.active?(url, exact: true)
+          item_class = active || (url.split('/').length > level && @filename.start_with?(url)) ? 'nav-subgroup active' : 'nav-subgroup'
+          sub_group_markup << "<li class=\"#{item_class}\">"
+          sub_group_markup << '<div class="nav-subgroup-heading">'
+          sub_group_markup << '<i class="material-icons">arrow_right</i>'
+          sub_group_markup << "<a href=\"#{url}\">#{title}</a>"
+          sub_group_markup << '</div>'
+          sub_group_markup << '<ul class="nav-ul">'
+
+          if grandchildren.empty?
+            headers.each do |header|
+              hash = header[:hash]
+              subtitle = header[:title]
+              sub_group_markup << "<li class=\"nav-leaf\"><a href=\"#{url}#{hash}\">#{subtitle}</a></li>"
+            end
+          else
+            sub_group_leaf_class = active ? 'nav-leaf nav-subgroup-leaf active' : 'nav-leaf nav-subgroup-leaf'
+            sub_group_markup << "<li class=\"#{sub_group_leaf_class}\"><a href=\"#{url}\">#{title} overview</a></li>"
+
+            grandchildren.each do |grandchild_path, grandchild_page|
+              sub_group_markup << generate_child_markup(grandchild_path, grandchild_page, grandchildren, 3)
+            end
+          end
+
+          sub_group_markup << '</ul>'
+          sub_group_markup << '</li>'
+        else
+          headers.each do |header|
+            hash = header[:hash]
+            subtitle = header[:title]
+            sub_group_markup << "<li class=\"nav-leaf\"><a href=\"#{url}#{hash}\">#{subtitle}</a></li>"
+          end
+        end
+      else
+        sub_group_markup << if has_child_pages
+                              "<li class=\"nav-leaf nav-subgroup-leaf\"><a href=\"#{url}\">#{title}</a></li>"
```

Comment:
Codacy found an issue: [Line is too long. [109/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)